packages/patternfly-4/react-core/src/components/ApplicationLauncher/ApplicationLauncherMenu.js

```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+import styles from '@patternfly/patternfly/components/AppLauncher/app-launcher.css';
+import { css } from '@patternfly/react-styles';
+import PropTypes from 'prop-types';
+import { ApplicationLauncherPosition } from './applicationLauncherConstants';
+import { DropdownContext, DropdownArrowContext } from '../Dropdown/dropdownConstants';
+import ReactDOM from 'react-dom';
+
+const propTypes = {
+  /** Anything which can be rendered as dropdown items */
+  children: PropTypes.node,
+  /** Classess applied to root element of dropdown menu */
+  className: PropTypes.string,
+  /** Flag to indicate if menu is opened */
+  isOpen: PropTypes.bool,
+  /** Indicates where menu will be alligned horizontally */
+  position: PropTypes.oneOf(Object.values(ApplicationLauncherPosition)),
+  /** Additional props are spread to the container component */
+  '': PropTypes.any
+};
+
+const defaultProps = {
+  children: null,
+  className: '',
+  isOpen: true,
+  position: ApplicationLauncherPosition.left,
+};
+
+class ApplicationLauncherMenu extends React.Component {
+  refsCollection = {};
+
+  keyHandler = (index, position, custom = false) => {
+    const kids = this.props.children;
+    if (!Array.isArray(kids)) return;
+    let nextIndex;
+    if (position === 'up') {
+      if (index === 0) {
+        // loop back to end
+        nextIndex = kids.length - 1;
+      } else {
+        nextIndex = index - 1;
+      }
+    } else if (index === kids.length - 1) {
+      // loop back to beginning
+      nextIndex = 0;
+    } else {
+      nextIndex = index + 1;
+    }
+    if (this.refsCollection[`item-${nextIndex}`] === null) {
+      this.keyHandler(nextIndex, position, custom);
+    } else {
+      custom
+        ? (this.refsCollection[`item-${nextIndex}`].current.focus &&
+            this.refsCollection[`item-${nextIndex}`].current.focus()) ||
+          ReactDOM.findDOMNode(this.refsCollection[`item-${nextIndex}`].current).focus()
+        : this.refsCollection[`item-${nextIndex}`].focus();
+    }
+  };
+
+  onKeyDown = event => {
```

Comment:
Can we make a constant in the helpers/constants files for the string "up", "ArrowUp" etc.  This keys will be used in multiple places.

Can we also create a helper function in our utils.ts file for the keyHandler.  It is used in multi components.  Make sure that you comment that the custom argument is optional.